packages/contracts/test/utils/version-comparison-lib.ts

```diff
@@ -0,0 +1,196 @@
+import {expect} from 'chai';
+import {ethers} from 'hardhat';
+
+import {
+  VersionComparisonLibTest,
+  VersionComparisonLibTest__factory,
+} from '../../typechain';
+
+type SemVer = [number, number, number];
+
+async function eqChecks(
+  func: (lhs: SemVer, rhs: SemVer) => Promise<boolean>,
+  expected: boolean
+) {
+  const results: boolean[] = await Promise.all([
+    func([1, 1, 1], [1, 1, 1]),
+    //
+    func([0, 1, 1], [0, 1, 1]),
+    func([1, 0, 1], [1, 0, 1]),
+    func([1, 1, 0], [1, 1, 0]),
+    //
+    func([1, 0, 0], [1, 0, 0]),
+    func([0, 1, 0], [0, 1, 0]),
+    func([0, 0, 1], [0, 0, 1]),
+    //
+    func([0, 0, 0], [0, 0, 0]),
+  ]);
+
+  // Check that all results match the expected value
+  expect(results.every(v => v === expected)).to.be.true;
+}
+
+async function ltChecks(
+  func: (lhs: SemVer, rhs: SemVer) => Promise<boolean>,
+  expected: boolean
+) {
+  const results: boolean[] = await Promise.all([
+    func([1, 1, 1], [2, 1, 1]),
+    func([1, 1, 1], [1, 2, 1]),
+    func([1, 1, 1], [1, 1, 2]),
+    //
+    func([1, 1, 1], [1, 2, 2]),
+    func([1, 1, 1], [2, 1, 2]),
+    func([1, 1, 1], [2, 2, 1]),
+    //
+    func([1, 1, 1], [2, 2, 2]),
+    //
+    func([1, 1, 0], [1, 2, 0]),
+    func([1, 1, 0], [2, 1, 0]),
+    //
+    func([1, 1, 0], [2, 2, 0]),
+    //
+    func([0, 1, 1], [0, 1, 2]),
+    func([0, 1, 1], [0, 2, 1]),
+    //
+    func([0, 1, 1], [0, 2, 2]),
+    //
+    func([1, 0, 0], [2, 0, 0]),
+    func([0, 1, 0], [0, 2, 0]),
+    func([0, 0, 1], [0, 0, 2]),
+  ]);
+
+  // Check that all results match the expected value
+  expect(results.every(v => v === expected)).to.be.true;
+}
+
+async function gtChecks(
+  func: (lhs: SemVer, rhs: SemVer) => Promise<boolean>,
+  expected: boolean
+) {
+  const results: boolean[] = await Promise.all([
+    func([2, 1, 1], [1, 1, 1]),
+    func([1, 2, 1], [1, 1, 1]),
+    func([1, 1, 2], [1, 1, 1]),
+    //
+    func([1, 2, 2], [1, 1, 1]),
+    func([2, 1, 2], [1, 1, 1]),
+    func([2, 2, 1], [1, 1, 1]),
+    //
+    func([2, 2, 2], [1, 1, 1]),
+    //
+    func([1, 2, 0], [1, 1, 0]),
+    func([2, 1, 0], [1, 1, 0]),
+    func([2, 2, 0], [1, 1, 0]),
+    //
+    func([0, 1, 2], [0, 1, 1]),
+    func([0, 2, 1], [0, 1, 1]),
+    func([0, 2, 2], [0, 1, 1]),
+    //
+    func([2, 0, 0], [1, 0, 0]),
+    func([0, 2, 0], [0, 1, 0]),
+    func([0, 0, 2], [0, 0, 1]),
+  ]);
+
+  // Check that all results match the expected value
+  expect(results.every(v => v === expected)).to.be.true;
+}
+
+describe('VersionComparisonLib', function () {
+  let cmp: VersionComparisonLibTest;
+
+  before(async () => {
+    const signers = await ethers.getSigners();
+    cmp = await new VersionComparisonLibTest__factory(signers[0]).deploy();
+  });
+
+  describe('eq', async () => {
+    async function eq(lhs: SemVer, rhs: SemVer): Promise<boolean> {
+      return await cmp.eq(lhs, rhs);
+    }
```

Comment:
```suggestion
    function eq(lhs: SemVer, rhs: SemVer): Promise<boolean> {
      return cmp.eq(lhs, rhs);
    }
```

Reply:
Fixed in 5568fe9d40f8032c78e181e68ff53a067013a455